pkg/api/v20240812preview/openshiftcluster_validatestatic.go

```diff
@@ -442,6 +443,14 @@ func (sv openShiftClusterStaticValidator) validatePlatformWorkloadIdentityProfil
 			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, fmt.Sprintf("%s.PlatformWorkloadIdentities[%d].resourceID", path, n), "Resource must be a user assigned identity.")
 		}
 	}
+
+	if pwip.UpgradeableTo != nil {
+		matches, err := regexp.MatchString(`^4\.[0-9]{2}\.`, string(*pwip.UpgradeableTo))
```

Comment:
You can use the CoreOS `semver` package here instead of rolling your own regex: https://pkg.go.dev/github.com/coreos/go-semver/semver#NewVersion. Just call this function and check for an error.

We already have this package vendored in, so no new imports needed.

Reply:
Nice, thanks for the tip.